### PR TITLE
IGNITE-8516 checked not null constraint

### PR DESCRIPTION
**Pull request by dgarus**

### CODE REVIEW

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCachePutNotNullFieldTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                return null;
+            },
+            IgniteSQLException.class
+        );
+    }
+
+    /** */
+    private static class Address implements Serializable {
+        /** */
+        private final String addr;
```
Comment (Mmuzaf) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCachePutNotNullFieldTest.java]: `.addQueryField("address", "java.lang.String", "address")`
and 
`private final String addr;`
from my point of view should have the same valuable name.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCachePutNotNullFieldTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ */
+public class IgniteCachePutNotNullFieldTest extends GridCommonAbstractTest {
+    /**
+     * @throws Exception If failed.
+     */
+    @SuppressWarnings("ThrowableNotThrown")
+    public void testPutAllShouldThrowExceptionWhenPassedNullValue() throws Exception {
+        final IgniteCache<Integer, Address> cache = jcache(
+            startGrid(0),
```
Comment (voipp) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCachePutNotNullFieldTest.java]: do you stop all grids after test execution ?

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCachePutNotNullFieldTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import org.apache.ignite.cache.CacheAtomicityMode;
+import org.apache.ignite.cache.QueryEntity;
+import org.apache.ignite.internal.processors.query.IgniteSQLException;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+
+/**
+ *
+ */
+public class IgniteCachePutNotNullFieldTest extends GridCommonAbstractTest {
```
Comment (voipp) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/IgniteCachePutNotNullFieldTest.java]: I think u should put test into IgniteSqlNotNullConstraintTest

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/atomic/GridNearAtomicUpdateFuture.java [hunk 4]
```diff
@@ -1105,8 +1105,12 @@ else if (conflictRmvVals != null) {
 
         KeyCacheObject cacheKey = cctx.toCacheKeyObject(key);
 
-        if (op != TRANSFORM)
+        if (op != TRANSFORM) {
             val = cctx.toCacheObject(val);
+
+            if (op == CREATE || op == UPDATE)
+                cctx.validateKeyAndValue(cacheKey, (CacheObject)val);
```
Comment (voipp) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/atomic/GridNearAtomicUpdateFuture.java]: it seems you should also add check in local cache

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlNotNullConstraintTest.java [hunk 5]
```diff
@@ -983,7 +959,7 @@ public void testReadThroughRestrictionCreateTable() throws Exception {
         GridTestUtils.assertThrowsAnyCause(log, new Callable<Object>() {
             @Override public Object call() throws Exception {
                 return executeSql("CREATE TABLE test(id INT PRIMARY KEY, name char NOT NULL) " +
-                    "WITH \"template=" + CACHE_READ_THROUGH+ "\"");
+                    "WITH \"template=" + CACHE_READ_THROUGH + "\"");
```
Comment (voipp) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/IgniteSqlNotNullConstraintTest.java]: do we need this space ?

**Pull request closed without merging**